src/workerd/api/system-streams.c++

```diff
@@ -237,12 +237,12 @@ kj::Promise<void> EncodedAsyncOutputStream::end() {
     promise = (*gz)->end().attach(kj::mv(*gz));
   }
 
-  KJ_IF_MAYBE(stream, inner.tryGet<kj::Own<kj::AsyncOutputStream>>()) {
-    if (auto casted = dynamic_cast<kj::AsyncIoStream*>(stream->get())) {
-      casted->shutdownWrite();
-    }
-    promise = promise.attach(kj::mv(*stream));
-  }
+  // KJ_IF_MAYBE(stream, inner.tryGet<kj::Own<kj::AsyncOutputStream>>()) {
+  //   if (auto casted = dynamic_cast<kj::AsyncIoStream*>(stream->get())) {
+  //     casted->shutdownWrite();
+  //   }
+  //   promise = promise.attach(kj::mv(*stream));
+  // }
```

Comment:
This causes the underlying AsyncIoStream to be closed whenever the WritableStream is closed. I'm not sure what the best solution to avoid this is. I know we added this code as part of https://github.com/cloudflare/workerd/pull/237 but I don't recall the reason for it.